crates/relayer-types/src/core/ics04_channel/packet.rs

```diff
@@ -178,6 +199,48 @@ impl Packet {
 
         height_timed_out || timestamp_timed_out
     }
+
+    /// If this packet data is an ICS-20 packet, check whether or not the memo and receiver
+    /// fields are under the given max size.
+    /// Returns `true` if this packet data is not an ICS-20 or fails to decode as an ICS-20 packet.
+    /// Return a ValidationResult::DecodeFailure this packet data is not an ICS-20 or fails to decode
+    /// as an ICS-20 packet.
+    /// Returns ValidationResult::Valid if both the memo and receiver field are valid.
+    /// Returns ValidationResult containing information on the failed fields if one or both fields fail
+    /// the validation.
+    pub fn are_fields_valid(
```

Comment:
Let's pull this out of the `Packet` struct as it is specific to ICS-20 and have it instead within the ics20 module as a free-standing function, perhaps named `check_ics20_fields_length`